java/src/jmri/jmrit/display/logixng/ActionLayoutTurnout.java

```diff
@@ -0,0 +1,499 @@
+import jmri.jmrit.logixng.util.parser.*;
+import jmri.jmrit.logixng.util.parser.ExpressionNode;
+import jmri.jmrit.logixng.util.parser.RecursiveDescentParser;
+import jmri.util.ThreadingUtil;
+import jmri.util.TypeConversionUtil;
+
+/**
+ * This action controls various things of a LayoutTurnout on a LayoutEditor panel.
+ *
+ * @author Daniel Bergqvist Copyright 2022
+ */
+public class ActionLayoutTurnout extends AbstractDigitalAction implements VetoableChangeListener {
+
+    private String _layoutEditorName;
+    private LayoutEditor _layoutEditor;
+    private NamedBeanAddressing _addressing = NamedBeanAddressing.Direct;
+    private LayoutTurnout _layoutTurnout;
+    private String _reference = "";
+    private String _localVariable = "";
+    private String _formula = "";
+    private ExpressionNode _expressionNode;
+    private NamedBeanAddressing _stateAddressing = NamedBeanAddressing.Direct;
+    private Operation _operation = Operation.Enable;
+    private String _stateReference = "";
+    private String _stateLocalVariable = "";
+    private String _stateFormula = "";
+    private ExpressionNode _stateExpressionNode;
+
+    public ActionLayoutTurnout(String sys, String user)
+            throws BadUserNameException, BadSystemNameException {
+        super(sys, user);
+    }
+
+    @Override
+    public Base getDeepCopy(Map<String, String> systemNames, Map<String, String> userNames) throws ParserException {
+        DigitalActionManager manager = InstanceManager.getDefault(DigitalActionManager.class);
+        String sysName = systemNames.get(getSystemName());
+        String userName = userNames.get(getSystemName());
+        if (sysName == null) sysName = manager.getAutoSystemName();
+        ActionLayoutTurnout copy = new ActionLayoutTurnout(sysName, userName);
+        copy.setComment(getComment());
+        copy.setLayoutEditor(_layoutEditorName);
+        copy.setLayoutTurnout(_layoutTurnout);
+        copy.setOperation(_operation);
+        copy.setAddressing(_addressing);
+        copy.setFormula(_formula);
+        copy.setLocalVariable(_localVariable);
+        copy.setReference(_reference);
+        copy.setStateAddressing(_stateAddressing);
+        copy.setStateFormula(_stateFormula);
+        copy.setStateLocalVariable(_stateLocalVariable);
+        copy.setStateReference(_stateReference);
+        return manager.registerAction(copy);
+    }
+
+    public void setLayoutEditor(@CheckForNull String layoutEditorName) {
+        assertListenersAreNotRegistered(log, "setEditor");
+        _layoutEditorName = layoutEditorName;
+        if (layoutEditorName != null) {
+            _layoutEditor = jmri.InstanceManager
+                    .getDefault(jmri.jmrit.display.EditorManager.class)
+                    .get(LayoutEditor.class, layoutEditorName);
+        } else {
+            _layoutEditor = null;
+        }
+//        InstanceManager.turnoutManagerInstance().addVetoableChangeListener(this);
```

Comment:
Is it a problem if the layout editor panel has been deleted?

Reply:
I have added a listener that listens to `EditorManager` and clears the reference to the editor and layout turnout if the editor is deleted.